fgpyo/util/inspect.py

```
@@ -328,14 +328,14 @@ def get_parser() -> partial:  # noqa: C901
         nonlocal type_
         nonlocal parsers
         try:
-            return functools.partial(parsers[type_])
+            return functools.partial(parsers[type_])  # type: ignore[index]
         except KeyError as ex:
             if (
                 type_ in [str, int, float]
                 or isinstance(type_, type)
                 and issubclass(type_, PurePath)
```

Ruff (0.15.12)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)